### PR TITLE
[#IOCIT-145] Upgraded to be compliant with functions v4

### PR DESCRIPTION
**Pull request by michaeldisaro**
#### Types of changes

- [ ] Chore (nothing changes by a user perspective)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -24,7 +24,7 @@
     "version": "auto-changelog -p --config .auto-changelog.json --unreleased && git add CHANGELOG.md"
   },
   "devDependencies": {
-    "@azure/functions": "^1.2.2",
+    "@azure/functions": "^3.2.0",
```
Comment (arcogabbo) [inline package.json:27]: should we also update `durable-functions` dependency? we are currently using a much outdated version.
I also see that `azure-storage` is now deprecated. To be more consistent with the runtime update I think we should update those dependencies.

File: docker/functions/Dockerfile [hunk 2]
```diff
@@ -2,7 +2,7 @@ FROM node:14.16.0
 
 WORKDIR /usr/src/app
 
-RUN npm i -g azure-functions-core-tools@3 --unsafe-perm true
+RUN npm i -g azure-functions-core-tools@4 --unsafe-perm true
```
Comment (arcogabbo) [inline docker/functions/Dockerfile:5]: I think this change needs to be done also in `docker/functions/Dockerfile` file?

File: package.json (hunk 1 above)
Comment (michaeldisaro) [reply package.json:27]: Yes, I think we must update also those, and be very careful! Anyway I'm not seeing any durable function in this repo, I'd check if the import is really needed.

File: yarn.lock [hunk 3]
```diff
@@ -57,11 +57,16 @@
     universal-user-agent "^6.0.0"
     uuid "^8.3.0"
 
-"@azure/functions@^1.2.2", "@azure/functions@^1.2.3":
+"@azure/functions@^1.2.3":
```
[the remark this answers is not in the document]
Comment (michaeldisaro) [reply yarn.lock:60]: As per above answer, I'll check if the import is really necessary.

**Review by arcogabbo — APPROVED** [review]
LGTM

**Pull request merged**